src/MagnumPlugins/AssimpImporter/AssimpImporter.cpp

```diff
@@ -1249,9 +1387,111 @@ Containers::Optional<MaterialData> AssimpImporter::doMaterial(const UnsignedInt
+                        type = MaterialAttributeType::Vector4i;
+                    else {
+                        Warning{} << "Trade::AssimpImporter::material(): property" << key << "is an integer array of" << property.mDataLength/4 << "items, saving as a typeless buffer";
+                        /* Using Pointer to indicate this is a buffer. String
+                           would indicate aiString which needs special handling
+                           for the length prefix. This still becomes an owned
+                           String later. */
+                        type = MaterialAttributeType::Pointer;
+                    }
+                } else if(property.mType == aiPTI_Float) {
+                    if(property.mDataLength/4 == 1)
+                        type = MaterialAttributeType::Float;
+                    else if(property.mDataLength/4 == 2)
+                        type = MaterialAttributeType::Vector2;
+                    else if(property.mDataLength/4 == 3)
+                        type = MaterialAttributeType::Vector3;
+                    else if(property.mDataLength/4 == 4)
+                        type = MaterialAttributeType::Vector4;
+                    else {
+                        Warning{} << "Trade::AssimpImporter::material(): property" << key << "is a float array of" << property.mDataLength/4 << "items, saving as a typeless buffer";
+                        /* See comment above */
+                        type = MaterialAttributeType::Pointer;
+                    }
+                } else if(property.mType == aiPTI_Double) {
+                    /** @todo This shouldn't happen without compiling Assimp
+                        with double support. But then the importer would only
+                        produce garbage because we assume ai_real equals float
+                        everywhere. */
+                    Warning{} << "Trade::AssimpImporter::material():" << key << "is a double precision property, saving as a typeless buffer";
+                    /* See comment above */
+                    type = MaterialAttributeType::Pointer;
+                } else if(property.mType == aiPTI_Buffer) {
+                    if(property.mDataLength == 1) {
+                        /* The glTF importer writes bool properties as buffers
+                           with size 1, when all the other importers write them
+                           as ints. Currently there is no importer that
+                           writes opaque buffer data, so we try to detect this.
+                           Funny library. */
+                        type = MaterialAttributeType::Bool;
+                    } else {
+                        /* See comment about Pointer further above */
+                        type = MaterialAttributeType::Pointer;
+                    }
+                } else if(property.mType == aiPTI_String) {
+                    type = MaterialAttributeType::String;
```

Comment:
I just tried that but it gets kinda awkward, since I have to define all the `type`, `value`, etc. variables earlier in the scope just for that one branch, and then later I still have to check for `MaterialAttributeType::String` to not call `materialAttributeTypeSize` which would otherwise assert.